src/msemu.c

```diff
@@ -58,11 +42,108 @@ unsigned char hex2bcd (unsigned char x)
 	return y;
 }
 
+/* Can be called multiple times, will zero the buffer if *io_buf is not null */
+/* Standard Z80 IO access will use lower 8bit of ADDR to be IO address with
+ * upper 8bit of ADDR being the value to write to the IO port
+ */
+int io_init(uint8_t **io_buf)
+{
+	if (*io_buf == NULL) {
+		*io_buf = (uint8_t *)calloc(SZ_256, sizeof(uint8_t));
+		if (*io_buf == NULL) {
+			printf("Unable to allocate IO buffer\n");
+			exit(EXIT_FAILURE);
+		}
+	} else {
+		/* Buffer is already allocated, just zero it out */
+		bzero(*io_buf, SZ_256);
+	}
+
+	return 0;
+}
+
+int io_deinit(uint8_t **io_buf)
+{
+	assert(*io_buf != NULL);
+	free(*io_buf);
+	return 0;
+}
+
+uint8_t io_read(uint8_t *io_buf, unsigned int absolute_addr)
+{
+	return *(io_buf + absolute_addr);
+}
+
+int io_write(uint8_t *io_buf, unsigned int absolute_addr, uint8_t val)
+{
+	*(io_buf + absolute_addr) = val;
+}
+
+/* Can be called multiple times, will zero the buffer if *ram_buf is not null */
+int ram_init(uint8_t **ram_buf)
+{
+	if (*ram_buf == NULL) {
+		*ram_buf = (uint8_t *)calloc(SZ_128K, sizeof(uint8_t));
+		if (*ram_buf == NULL) {
+			printf("Unable to allocate RAM buffer\n");
+			exit(EXIT_FAILURE);
+		}
+	} else {
+		/* Buffer is already allocated, just zero it out */
+		bzero(*ram_buf, SZ_128K);
```

Comment:
> The bzero() function is deprecated (marked as LEGACY in
       POSIX.1-2001); use memset(3) in new programs.

https://man7.org/linux/man-pages/man3/bzero.3.html

Reply:
Oh, huhn, I did not realize that. bzero() is used in a few places, will update.

Reply:
Changed in the few places it was used.